src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -63,15 +63,45 @@ void AsynchronousMetrics::run()
 {
     setThreadName("AsyncMetrics");
 
-    /// Next minute + 30 seconds. To be distant with moment of transmission of metrics, see MetricsTransmitter.
-    const auto get_next_minute = []
+    const auto get_next_update_time = []
     {
-        return std::chrono::time_point_cast<std::chrono::minutes, std::chrono::system_clock>(
-            std::chrono::system_clock::now() + std::chrono::minutes(1)) + std::chrono::seconds(30);
+        using namespace std::chrono;
+
+        // The period doesn't really have to be configurable, but sometimes you
+        // need to change it by recompilation to debug something. The generic
+        // code is left here so that you don't have to ruin your mood by touching
+        // std::chrono.
+        const seconds period(60);
+
+        const auto now = time_point_cast<seconds>(system_clock::now());
+
+        // Use seconds since the start of the hour, because we don't know when
+        // the epoch started, maybe on some weird fractional time.
+        const auto start_of_hour = time_point_cast<seconds>(time_point_cast<hours>(now));
```

Comment:
It would be unusual if we use period like 7 seconds.
It is not important, I suppose.